packages/KDateTimePicker/KDateTimePicker.vue

```diff
@@ -0,0 +1,502 @@
+          start,
+          end,
+          startISO: new Date(start).toISOString(),
+          endISO: (end !== '') ? new Date(end).toISOString() : end
+        }
+
+        this.abbreviatedDisplay = this.formatDisplayDate(range)
+      }
+    }
+  },
+
+  mounted () {
+    // Select the tab based on incoming defaults
+    if (this.hasDefaultCustomValue) {
+      this.tabName = 'custom'
+    } else if (this.hasRelativeTimeframes) {
+      this.tabName = 'relative'
+    }
+
+    // Set default value to be displayed in the input field
+    if (this.hasDefaultCustomValue) {
+      const range = {
+        start: this.defaultCustom.start,
+        end: this.defaultCustom.end
+      }
+
+      this.abbreviatedDisplay = this.formatDisplayDate(range)
+    } else if (this.defaultTimeframe) {
+      this.changeRelativeTimeframe(this.defaultTimeframe)
+    }
+  },
+
+  methods: {
+    getText (tf) {
+      return tf.timeframeText
+    },
+
+    /**
+     * Updates both the input field value, and the full time frame readout
+     * when a relative time frame button is clicked
+     * @param {*} timeframe
+     */
+    changeRelativeTimeframe (timeframe) {
+      this.selectedTimeframe = timeframe
+
+      // Update input field text
+      this.abbreviatedDisplay = this.selectedTimeframe.timeframeText
+
+      // Format the start/end values as human readable date
+      const start = this.selectedTimeframe.start()
+      const end = this.selectedTimeframe.end()
+
+      // Set value to be emitted when relative time frame clicked
+      this.selectedRange = {
+        start,
+        end,
+        startISO: new Date(start).toISOString(),
+        endISO: new Date(end).toISOString(),
+        timeframeText: this.selectedTimeframe.timeframeText
+      }
+
+      this.fullRangeDisplay = this.formatDisplayDate({ start, end })
+    },
+
+    clearSelection () {
+      this.selectedCalendarRange = null
+      this.abbreviatedDisplay = this.defaultMessage
+      this.fullRangeDisplay = ''
+      this.selectedRange = { start: '', end: '' }
+      this.selectedTimeframe = this.timePeriods[0]
+    },
+
+    // Displays selected date/time/range as a human readable string
+    formatDisplayDate (range) {
+      const { start, end } = range
+      let fmtStr = 'PP'
+      let fmtRange = ''
+
+      // calendarMode determines format string
+      if (this.calendarMode === 'date') {
+        fmtStr = 'PP'
+      } else if (this.calendarMode === 'time') {
+        fmtStr = 'PP hh:mm a'
+      } else if (this.calendarMode === 'dateTime') {
+        fmtStr = 'PP hh:mm a'
+      }
+
+      // isRange determines whether an end date/time should be shown
+      if (this.isRange || this.hasRelativeTimeframes) {
+        fmtRange = `${format(start, fmtStr)} - ${format(end, fmtStr)}`
+      } else {
+        fmtRange = `${format(start, fmtStr)}`
+      }
+
+      return fmtRange
+    },
+
+    getDefaultTabName () {
+      return (this.hasDefaultCustomValue ? 'custom' : 'relative')
```

Comment:
Don't need parenthesis
```suggestion
      return this.hasDefaultCustomValue ? 'custom' : 'relative'
```

Reply:
Haters gon' hate 🤠 

Call me old school, but I think of these parantheses as a soft, snuggly blanket that provides security and comfort to an otherwise floaty (and lonely) piece of logic.

🤣

Reply:
😴 <- this is cozy Mihai 😂